common/src/main/java/tachyon/util/network/NetworkAddressUtils.java

```diff
@@ -241,10 +337,164 @@ public static ServerSocket getSocket(final TServerSocket thriftSocket) {
    * @return the InetSocketAddress of the master
    */
   public static InetSocketAddress getMasterAddress(TachyonConf conf) {
-    String masterHostname =
-        conf.get(Constants.MASTER_HOSTNAME, getLocalHostName(conf));
+    String masterHost = conf.get(Constants.MASTER_HOSTNAME, getLocalHostName(conf));
     int masterPort = conf.getInt(Constants.MASTER_PORT, Constants.DEFAULT_MASTER_PORT);
-    return new InetSocketAddress(masterHostname, masterPort);
+    return new InetSocketAddress(masterHost, masterPort);
+  }
+
+  /**
+   * Helper method to get the {@link InetSocketAddress} connect address of the master.
+   *
+   * @param conf the configuration of Tachyon
+   * @return the master's connect address
+   */
+  public static InetSocketAddress getMasterConnectAddress(TachyonConf conf) {
+    if (conf.get(Constants.MASTER_BIND_HOST, "localhost").equals("0.0.0.0")) {
+      String masterHostName = getMasterHostName(conf);
+      int masterPort = conf.getInt(Constants.MASTER_PORT, Constants.DEFAULT_MASTER_PORT);
+      TachyonConf.assertValidPort(masterPort, conf);
+      return new InetSocketAddress(masterHostName, masterPort);
+    }
+    return getMasterBindAddress(conf);
+  }
+
+  /**
+   * Gets the Tachyon master address from the configuration
+   *
+   * @param conf the configuration of Tachyon
+   * @return the InetSocketAddress of the master
+   */
+  public static InetSocketAddress getMasterBindAddress(TachyonConf conf) {
```

Comment:
I'm a bit confused as to the difference between the `get*BindAddress` and `get*ConnectAddress` variants of methods. What is the difference between a "bind" address and a "connect" address?